roles/translator/src/main.rs

```diff
@@ -161,9 +162,26 @@ async fn main() {
         b,
     );
 
+    let mut interrupt_signal_future = Box::pin(tokio::signal::ctrl_c().fuse());
+
     // Check all tasks if is_finished() is true, if so exit
     loop {
-        let task_status = rx_status.recv().await.unwrap();
+        let task_status = select! {
+            task_status = rx_status.recv().fuse() => task_status,
```

Comment:
Can you add a comment describing what this select! is doing? Just something simple like "this code is waiting for either a value to be received on the rx_status channel or for an interrupt signal to be received and then breaking the main loop to end the process if appropriate"

Reply:
Hmm... to me, comments are about explaining why code is doing what it is doing, when it isn't clear why. I'm happy to defer to you if you feel it's valuable here, though.

Reply:
Yeah, that's fair! This is pretty self explanatory so I'm fine with no comment.